### ISSUE TITLE
Crypto Fail for Credential Storage

### ISSUE THREAD

**Issue by secretsquirrel** [jitsi/jitsi#122]
Static value for creating key:
https://github.com/jitsi/jitsi/blob/master/src/net/java/sip/communicator/impl/credentialsstorage/AESCrypto.java#L29

Static value for salt in creating key:
https://github.com/jitsi/jitsi/blob/master/src/net/java/sip/communicator/impl/credentialsstorage/AESCrypto.java#L39

ECB...
https://github.com/jitsi/jitsi/blob/master/src/net/java/sip/communicator/impl/credentialsstorage/AESCrypto.java#L34

**Comment by cobratbq** [comment]
I don't get your first point. I am assuming that the emphasis is on "**static** value for creating key". Are you suggesting that this should be determined at the moment of use?
If so, that would mean that with a change of JVM or maybe even available security extensions, one would not be able to decrypt his passwords anymore.

So, could you elaborate on this some more?

**Comment by secretsquirrel** [comment]
I'm guessing you are trying to protect credentials at rest (on disk)?

**Comment by cobratbq** [comment]
I am not familiar with this code, so what I am saying now is an educated guess at best. I suspect this code is used for encrypting and decrypting passwords that are being stored in protected form in Jitsi's configuration file.

**Comment by secretsquirrel** [comment]
Yeah. Typically you would want to use a user provided key, a random salt, and not AES-ECB.

**Comment by cobratbq** [comment]
I understand your 2nd and 3rd point. There are better alternatives for ECB and typically you generate a random salt that you persist with the other data. However, your first reference is not the key itself, it is the key algorithm that was used while generating the secret data.
So, that why I asked the question: what is your point for your first reference: "Static value for creating key"?

See http://docs.oracle.com/javase/7/docs/api/javax/crypto/spec/SecretKeySpec.html#SecretKeySpec%28byte[],%20int,%20int,%20java.lang.String%29

**Comment by secretsquirrel** [comment]
I see.  I made a mistake about the first item.  It's either NULL or a user provided password later -- on in the document.

**Comment by cobratbq** [comment]
Okay, sure, and your other 2 remarks I understand, so we're clear on the issue then.

**Comment by TheRealMurmel** [comment]
3 years left, are there any plans on fixing this issue? 
Especially on the first item (hardcoded key): I think it would be nice to at least document this behaviour anywhere visible for the user, so the user is aware of this. Otherwise this could create a false sense of security.
Any plans on using the keyring on linux based systems for storing the masterpassword?
